NewHorizons/Handlers/HeldItemHandler.cs

```diff
@@ -0,0 +1,229 @@
+using HarmonyLib;
+using NewHorizons.Builder.Props;
+using NewHorizons.External.Modules.Props;
+using NewHorizons.Utility;
+using NewHorizons.Utility.OWML;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+namespace NewHorizons.Handlers;
+
+[HarmonyPatch]
+public static class HeldItemHandler
+{
+    /// <summary>
+    /// Dictionary of system name to item path
+    /// If we travel to multiple systems within a single loop, this will hold the items we move between systems
+    /// </summary>
+    private static Dictionary<string, HashSet<string>> _pathOfItemTakenFromSystem = new();
+
+    public static bool WasAWCTakenFromATP => _pathOfItemTakenFromSystem.TryGetValue("SolarSystem", out var list) && list.Contains(ADVANCED_WARP_CORE);
+
+    private static GameObject _currentlyHeldItem;
+
+    /// <summary>
+    /// Track the path of any item we ever pick up, in case we take it out of the system
+    /// </summary>
+    private static Dictionary<GameObject, string> _trackedPaths = new();
+
+    private static bool _isInitialized = false;
+    private static bool _isSystemReady = false;
+
+    /// <summary>
+    /// We keep our own reference to this because when Unload gets called it might have already been updated
+    /// </summary>
+    private static string _currentStarSystem;
+
+    private const string ADVANCED_WARP_CORE = "TowerTwin_Body/Sector_TowerTwin/Sector_TimeLoopInterior/Interactables_TimeLoopInterior/WarpCoreSocket/Prefab_NOM_WarpCoreVessel";
+
+    [HarmonyPrefix, HarmonyPatch(typeof(ItemTool), nameof(ItemTool.Awake))]
+    private static void Init()
+    {
+        _trackedPaths.Clear();
+
+        _currentStarSystem = Main.Instance.CurrentStarSystem;
+
+        if (!_isInitialized)
+        {
+            _isInitialized = true;
+            Main.Instance.OnChangeStarSystem.AddListener(OnStarSystemChanging);
+            Main.Instance.OnStarSystemLoaded.AddListener(OnSystemReady);
+            GlobalMessenger<DeathType>.AddListener("PlayerDeath", OnPlayerDeath);
+        }
+    }
+
+    private static void OnPlayerDeath(DeathType _)
+    {
+        NHLogger.Log("Player died, resetting held items");
+
+        // Destroy everything
+        _pathOfItemTakenFromSystem.Clear();
+        GameObject.Destroy(_currentlyHeldItem);
+    }
+
+    private static GameObject MakePerfectCopy(GameObject go)
+    {
+        var owItem = go.GetComponent<OWItem>();
+
+        var tempParent = new GameObject();
+        tempParent.transform.position = new Vector3(100000, 0, 0);
+        var newObject = DetailBuilder.Make(tempParent, tempParent.AddComponent<Sector>(), null, go, new DetailInfo() { keepLoaded = true });
```

Comment:
Looks like tempParent never gets destroyed/disabled; is that an issue? I guess not since it's an empty sector and nothing else.